File: Decimal_to_roman/JAVA/src/dec_to_roman.java
public class dec_to_roman {
    private static final int[] VALUES = { 1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1 };
    private static final String[] SYMBOLS = { "M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I" };

    public String intToRoman(int num) {
        StringBuilder roman = new StringBuilder();
        for (int i = 0; i < VALUES.length; i++) {
            while (num >= VALUES[i]) {
                roman.append(SYMBOLS[i]);
                num -= VALUES[i];
            }
        }
        return roman.toString();
    }
}

File: Decimal_to_roman/CPP/src/dec_to_roman.cpp
#include "../include/dec_to_roman.hpp"
#include <vector>
#include <string>

std::string DecimalToRoman::intToRoman(int num) {
    // Define symbols for Roman numerals and their corresponding values
    std::vector<std::pair<int, std::string>> romanSymbols = {
        {1000, "M"}, {900, "CM"}, {500, "D"}, {400, "CD"},
        {100, "C"}, {90, "XC"}, {50, "L"}, {40, "XL"},
        {10, "X"}, {9, "IX"}, {5, "V"}, {4, "IV"}, {1, "I"}
    };

    std::string romanNumeral = "";
    
    // Iterate through the symbols and convert the number to Roman numerals
    for (const auto& symbol : romanSymbols) {
        int value = symbol.first;
        const std::string& romanSymbol = symbol.second;
        
        while (num >= value) {
            romanNumeral += romanSymbol;
            num -= value;
        }
    }

    return romanNumeral;
}
